packages/graphql-transformers-e2e-tests/src/__tests__/SubscriptionsWithAuthTest.e2e.test.ts

```diff
@@ -209,7 +209,8 @@ beforeAll(async () => {
 
     type Post @model
         @auth(rules: [
-            {allow: owner, ownerField: "postOwner"}
+            { allow: owner, ownerField: "postOwner" }
+            { allow: private, operations: [read], provider: iam }
```

Comment:
I think we should create a separate type for this instead of changing the cognito based auth type to include IAM.

Reply:
Adding this change tests that we can subscribe using IAM auth on a mutation that used UserPools auth. This also ensures we can call the subscription without an argument.

Reply:
But do we have any tests where owner based auth makes subscription parameter non-billable?

Reply:
good point added a unit test to confirm when only an owner rule is used the owner arg should be non nullable